packages/devtools_app/lib/src/screens/network/har_network_data.dart

```diff
@@ -55,3 +56,149 @@ class HarNetworkData with Serializable {
     return buildHar(requests);
   }
 }
+
+class HarDataEntry {
+  HarDataEntry(this.request);
+
+  /// Creates an instance of [HarDataEntry] from a JSON object.
+  ///
+  /// This factory constructor expects the [json] parameter to be a Map
+  /// representing a single HAR entry.
+  factory HarDataEntry.fromJson(Map<String, Object?> json) {
+    _convertHeaders(json);
+
+    final modifiedRequestData =
+        _remapCustomFieldKeys(json) as Map<String, dynamic>;
+
+    // Retrieving url, method from requestData
+    modifiedRequestData['uri'] =
+        (modifiedRequestData['request'] as Map<String, Object?>)['url'];
+    modifiedRequestData['method'] =
+        (modifiedRequestData['request'] as Map<String, Object?>)['method'];
+
+    // Adding missing keys which are mandatory for parsing
+    (modifiedRequestData['response'] as Map<String, Object?>)['redirects'] = [];
+    Object? requestPostData;
+    Object? responseContent;
+    if (modifiedRequestData['response'] != null &&
+        (modifiedRequestData['response'] as Map<String, Object?>)['content'] !=
+            null) {
+      responseContent =
+          (modifiedRequestData['response'] as Map<String, Object?>)['content'];
+    }
+
+    if (modifiedRequestData['request'] != null &&
+        (modifiedRequestData['request'] as Map<String, Object?>)['postData'] !=
+            null) {
+      requestPostData =
+          (modifiedRequestData['response'] as Map<String, Object?>)['content'];
+    }
+
+    return HarDataEntry(
+      DartIOHttpRequestData.fromJson(
+        modifiedRequestData,
+        requestPostData as Map<String, Object?>,
+        responseContent as Map<String, Object?>,
+      ),
+    );
+  }
+
+  final DartIOHttpRequestData request;
+
+  /// Converts the instance to a JSON object.
+  ///
+  /// This method returns a Map representing a single HAR entry, suitable for
+  /// serialization.
+  Map<String, Object?> toJson() {
+    // Implement the logic to convert DartIOHttpRequestData to HAR entry format
+    return {};
```

Comment:
Yes, however one doubt here, can we call buildHar here directly ? Because HarDataEntry represents a single entry, so shouldn't we segregate the entries and build each entry in the toJson of this class?
and in build har we write 
  ```dart final entries = httpRequests.map((e) => HarDataEntry.toJson(e)).toList();```
and call build har from HarNetworkData's toJson (which we are already doing).

Reply:
I have pushed this implementation here

Reply:
Oh okay, glad to hear it!